### PR TITLE
[autogluon][build] Updating scan configuration

### PR DESCRIPTION
**Pull request by gradientsky**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Adding exclusions until upstream dependencies are updated

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/inference/docker/0.4/py3/Dockerfile.cpu.os_scan_allowlist.json [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      "cvss_v3_score": 7.7,
+      "cvss_v30_score": 0.0,
+      "cvss_v31_score": 7.7,
+      "cvss_v2_score": 0.0,
+      "cvss_v3_severity": "HIGH",
+      "source_url": "https://snyk.io/vuln/SNYK-JAVA-COMGOOGLECODEGSON-1730327",
+      "source": "SNYK",
+      "severity": "HIGH",
+      "status": "ACTIVE",
+      "title": "SNYK-JAVA-COMGOOGLECODEGSON-1730327 - com.google.code.gson:gson"
```
Comment (shantanutrip) [inline autogluon/inference/docker/0.4/py3/Dockerfile.cpu.os_scan_allowlist.json:59]: Can we please add a key called `reason_to_ignore`: "xyz..."?

Comment (shantanutrip) [reply autogluon/inference/docker/0.4/py3/Dockerfile.cpu.os_scan_allowlist.json:59]: As seen here: https://github.com/aws/deep-learning-containers/pull/2328/files#diff-67be49f98eac4ee50ac13b54139abc30c2451fe163e30d876c3d09849c0a9a16R30

Comment (shantanutrip) [reply autogluon/inference/docker/0.4/py3/Dockerfile.cpu.os_scan_allowlist.json:59]: I just realized that we have added `reason_to_ignore` at the top. Will it be possible to place it at the bottom to maintain consistency while reading/reviewing?

File: autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -25,6 +25,9 @@ RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.p
 # Resolving sagemaker 2.92.1 conflict
 RUN pip install --no-cache-dir 'importlib-metadata<2.0,>=1.4.0' markdown
 
+# Removing GluonTS poetry.lock - creates false alarms on scans
+RUN rm -rf /usr/local/lib/python3.8/dist-packages/gluonts/nursery/tsbench/poetry.lock
```
Comment (shantanutrip) [inline autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu:29]: I had a second thought regarding the removal of this package. Is it possible that the customers could be relying on this for their usage?

Comment (shantanutrip) [reply autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu:29]: I am apprehensive that we might end up removing something that might cause failures for the customers. Had it been the very first release of a new AG version, this would not have been a problem.

Comment (gradientsky) [reply autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu:29]: tsbench is a benchmark tooling (not sure why it's bundled within the library). We don't expect any production users to use it as a part of normal use of AG.
I did some sanity tests for time series: it does not affect training/inference functionality.

File: autogluon/inference/docker/0.4/py3/Dockerfile.cpu.os_scan_allowlist.json (hunk 1 above)
Comment (gradientsky) [reply autogluon/inference/docker/0.4/py3/Dockerfile.cpu.os_scan_allowlist.json]: Updated `reason_to_ignore` to be at the bottom.

File: autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu (hunk 2 above)
Comment (shantanutrip) [reply autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu]: Got it, thanks!!

**Review by ohadkatz — APPROVED** [review]

**Comment by ohadkatz** [comment]
LGTM
